client/src/components/LaunchControls.tsx

```diff
@@ -24,49 +70,34 @@ export default function LaunchControls() {
     displayToStorageUnit: speedDisplayToStorageUnit,
     storageToDisplayUnit: speedStorageToDisplayUnit,
   } = useUnitHelper(MeasurementTypes.Speed);
-  const { start, reset } = useSimulationActions();
-
-  const roundedInitialSpeed = speedStorageToDisplayUnit(
-    launchConditions.speed
-  ).toFixed(2);
-  const roundedInitialVerticalAngle = angleStorageToDisplayUnit(
-    launchConditions.verticalAngle
-  ).toFixed(2);
-  const [speed, setSpeed] = useState(roundedInitialSpeed);
-  const [verticalAngle, setVerticalAngle] = useState(
-    roundedInitialVerticalAngle
-  );
+  const { updateLaunchConditions } = useSimulationActions();
 
   return (
-    <>
-      <TextInput
+    <Box w={200}>
+      <NumberInput
```

Comment:
The numeric input here handles rounding pretty nicely, so now we can pass the modified values directly to the simulation store